README.md

```diff
@@ -1 +1,285 @@
+  </ul>
+</details>
+
+
+#### constructor(parent)
+
+The constructor expects a `parent` pointing to the parent chain, if any.
+
+#### .end()
+
+Ends the current chain, going back to the parent chain.
+
+Returns the parent.
+
+#### .batch(fn)
+
+Execute a function against the current chain.   
+This is useful to execute various operations while not breaking the chain.
+
+Returns itself to allow chaining.
+
+#### .when(condition, whenTruthy, whenFalsy)
+
+Conditionally execute a function on the current chain.   
+`whenTruthy` and `whenFalsy` are called if condition is truthy or falsy respectively.
+
+Returns itself to allow chaining.
+
+
+### ChainedMap
+
+A ChainedMap operates similar to a JavaScript Map, with some conveniences for chaining and generating data structures. It extends [Chainable](#chainable) which means that all its methods are also available.
+
+<details>
+  <summary><b>Diferences from webpack-chain:</b></summary>
+  <ul>
+    <li>Removed <code>order()</code> because its an internal method, reducing the number of conflicts in case you create a class that inherits from <code>ChainedMap</code></li>
+    <li>Removed <code>.clean(obj)</code> because it's more of a helper (it didn't use <code>this.store</code> whatsoever)</li>
+    <li>Added <code>.tap(key, fn)</code></li>
+    <li>Added <code>.keys()</code> just like the native <a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/keys">Map</a></li>
+    <li>Added <code>.forEach()</code> just like the native <a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/forEach">Map</a></li>
+    <li>Changed <code>.entries()</code> to no longer return <code>undefined</code> if the map is empty in order to avoid any "gotchas"</li>
+  </ul>
+</details>
+
+
+#### .get(key)
+
+Retrieve the value of the entry corresponding to `key`.
+
+#### .set(key, value)
+
+Set the value the entry corresponding to `key`.
+
+Returns itself to allow chaining.
+
+#### .tap(key, fn)
+
+Execute `fn` with the current value of the entry corresponding to `key`.   
+The return value of `fn` will replace the current value.
+
+Returns itself to allow chaining.
+
+#### .delete(key)
+
+Remove entry corresponding to `key`.
+
+Returns itself to allow chaining.
+
+#### .has(key)
+
+Check if there's an entry corresponding to `key`.
+
+Returns `true` if it exists, `false` otherwise.
+
+#### .clear()
+
+Remove all entries from the Map.
+
+Returns itself to allow chaining.
+
+#### .keys()
+
+Returns an array of the keys stored in the Map.
+
+The order will having into consideration entries that called `.before(name)` or `.after(name)`, see [OrderableChainedMap](#orderable-chained-map).
+
+#### .values()
+
+Returns an array of the values stored in the Map.
+
+The order will having into consideration entries that called `.before(name)` or `.after(name)`, see [OrderableChainedMap](#orderable-chained-map).
```

Comment:
_The result will have into consideration the calls to `.before(name)` or `.after(name)`. See [OrderableChainedMap](#orderable-chained-map)._